lib/Backend/SccLiveness.cpp

```diff
@@ -53,24 +53,30 @@ SCCLiveness::Build()
             AssertMsg(instr->IsBranchInstr(), "Loop tail should be a branchInstr");
             AssertMsg(instr->AsBranchInstr()->IsLoopTail(this->func), "Loop tail not marked correctly");
 
-            FOREACH_SLIST_ENTRY(Lifetime *, lifetime, this->curLoop->regAlloc.extendedLifetime)
+            Loop *loop = this->curLoop;
+            while (loop && loop->regAlloc.loopEnd == this->curLoop->regAlloc.loopEnd)
```

Comment:
It is guaranteed to be extended now. I have added an Assert for parent's loopEnd to make sure of this.